pkg/webrtctransport.go

```diff
@@ -34,6 +34,7 @@ type WebRTCTransport struct {
 	mu             sync.RWMutex
 	session        *Session
 	routers        map[uint32]*Router
+	senders        map[uint32]*WebRTCSender
```

Comment:
we could iterate though the routers but would be a bit extra compute. it will also be useful for a transport to track its senders for simulcast. so the transport can modify the tracks it is receiving from a sender

Reply:
I think that the extra compute of iterating the routers it's negligible, but it's much better to have only one source of truth for the senders, so we don't have add/ delete senders in different places and maybe introduce race conditions.  If we keep this way there is only a missing method to delete senders in transports.

Besides that It looks good to me.

Reply:
yeah i agree. i will change it to iterate routers for now and we can figure out track switching later. i think there is still a race condition here where a sender is added between offer/answer. i haven't thought of a clean solution though. any ideas?

Reply:
Yes I was thinking that the reciever transport should not enable their senders, since the senders may be in different states. But senders should enable themselves when ready.

And a sender will be ready to recieve when:
 1. On join after setting answer. (Since now the user subscribes after joining only we need to focus on negotiation)
 2. On negotiation when answer is recieved from client, when negotiation is requested on server (after a track is added).

The problem is that when the first answer is recieved, there are going to be many senders from same client in different states, since sometimes a negotiation is required per track and per users in the conference. So we need to relate an aswer to a sender, this maybe expensive and difficult I guess. 

So maybe we can take a naive approach and just add a sleep of X seconds before sending packets to the client, until we found a better approach.

Reply:
hmm i prefer the existing race condition over the one introduced with the timeout approach. i'll think about this a bit more. i think we would need to track the new senders for each offer in a queue. 

another solution is we could use a data channel to communicate the ontrack event from the client. so it would notify the sfu when it has received a track and rtp should flow. the same data channel can be used for track switching in simulcast, ect.

Reply:
The data channels approach It's the most safe approach, and It's the way mediasoup solved the problem by signaling from client to server when ready.